### PR TITLE
Blogpost: doctrine optimization hydration

### PR DESCRIPTION
**Pull request by Ocramius**
Ping @malukenho

### CODE REVIEW

File: source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html [hunk 1]
```diff
@@ -0,0 +1,542 @@
+---
+layout: post
+title: Doctrine ORM Hydration Performance Optimization
+category: PHP
+tags: ["php", "doctrine", "orm", "hydration", "performance", "speed"]
+year: 2015
+month: 4
+day: 10
```
Comment (Ocramius) [inline source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html]: Date will need to be adjusted

File: source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+    generalize as <code>O(n ^ m)</code>, with <code>n</code> being the amount of results, and <code>m</code>
+    being the amount of joined tables.
+</p>
+
+<p>
+    Here is a graphical representation of <code>O(n ^ m)</code>:
+</p>
+
+<p>
+    <!-- Well that escalated quickly image here -->
```
Comment (Ocramius) [inline source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html]: Need silly meme here

File: source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+    ')
+    ->getResult(); // result is discarded (this is just re-hydrating the collections)
+
+return $users;
+~~~
+
+<p>
+    I'd also add that this is the only legitimate use-case for partial hydration that I ever
+    had, but it's a personal opinion/feeling.
+</p>
```
Comment (Ocramius) [inline source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html]: Needs a conclusion (which I still need to write)

File: source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html [hunk 4]
```diff
@@ -0,0 +1,542 @@
+---
```
Comment (Ocramius) [inline source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html]: Note: file will be renamed to `.md`

File: source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html [hunk 5]
```diff
@@ -0,0 +1,548 @@
...
+
+return $results;
+~~~
+
+<p>
+    That's a very basic example, but this gives you an idea of what an ORM is doing for you.
+</p>
+
+<p>
+    As you can see, this is an <code>O(N)</code> operation (assuming a constant number of reflection fields).
```
Comment (beberlei) [inline source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html]: this is O(N^2)

File: source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html [hunk 6]
```diff
@@ -0,0 +1,548 @@
...
+</p>
+
+<p>
+    The hydration operations on this resultset are <code>O(n * m * q)</code>, which I'm going to simply
+    generalize as <code>O(n ^ m)</code>, with <code>n</code> being the amount of results, and <code>m</code>
+    being the amount of joined tables.
+</p>
+
+<p>
+    Here is a graphical representation of <code>O(n ^ m)</code>:
```
Comment (beberlei) [inline source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html]: this should probably be O(N*M)

Comment (Ocramius) [inline source/_posts/2015-04-10-doctrine-orm-optimization-hydration.html]: As discussed on IRC, this is actually a pow() - `m` is the number of joins

**Pull request merged**